src/vs/workbench/browser/viewlet.ts

```diff
@@ -259,12 +259,12 @@ export class ToggleViewletAction extends Action {
 export class CollapseAction extends Action {
 
 	constructor(viewer: ITree, enabled: boolean, clazz: string) {
-		super('workbench.action.collapse', nls.localize('collapse', "Collapse All"), clazz, enabled, (context: any) => {
+		super('workbench.action.collapse', nls.localize('collapse', "Collapse"), clazz, enabled, (context: any) => {
 			if (viewer.getHighlight()) {
 				return TPromise.as(null); // Global action disabled if user is in edit mode from another action
 			}
 
-			viewer.collapseAll();
+			viewer.collapseDeepestExpandedLevel();
```

Comment:
I think we have two actions that do the same thing. Let's replace one with an action that calls `collapseDeepestExpandedLevel`, and leave the other. And let's be explicit in which trees were are changing the behavior of. We should change it for search results in this PR and maybe the file explorer. If we want other trees to have this behavior, we can change that later.

Reply:
I've scoped it to just search results for now